dataprep/clean/clean_df.py

```diff
@@ -0,0 +1,414 @@
+    atomic_data_types_df.columns = ["atomic_data_type"]
+    return atomic_data_types_df
+
+
+def _fill_missing_values_column(column: pd.Series) -> None:
+    """
+    Function to standardize the missing values for a column, depending on its data type detection:
+    1. For date, change to pd.NaT
+    3. For any other data type, change to np.nan
+
+    Parameters
+    ----------
+    column
+        Columns of a Pandas DataFrame to be standardized.
+    """
+
+    if infer_dtype(column) == "date":
+        column[column.apply(_check_null_values, 0)] = pd.NaT
+        column = pd.to_datetime(column)
+    else:
+        column[column.apply(_check_null_values, 0)] = np.nan
+
+
+def _standardize_missing_values_df(
+    df: pd.DataFrame, standardize_missing_values: str
+) -> pd.DataFrame:
+    """
+    Function to standardize the missing values for a whole dataframe.
+
+    Parameters
+    ----------
+    df
+        Pandas Dataframe to be standarized.
+    standardize_missing_values {‘fill’, ‘remove’, ’ignore’}
+        * If ’fill’, then all detected missing values will be set as np.NaN.
+        * If ‘remove’, then any rows with missing values will be deleted.
+        * If ‘ignore’, then no action will be taken.
+    """
+    if standardize_missing_values == "fill":
+        df.apply(_fill_missing_values_column, 0)
+        df = df.convert_dtypes()
+    elif standardize_missing_values == "remove":
+        df.apply(_fill_missing_values_column, 0)
+        df = df.convert_dtypes()
+        df.dropna(how="any", inplace=True)
+    else:
+        pass
+
+    return df
+
+
+def _create_report(stats: Dict[str, Any], old_stat: Any, option: str) -> None:
+    """
+    Describe what was done in the cleaning process.
+    """
+    if option == "datatype":
+        print("Data Type Detection Report:")
+        print(
+            "\tThese data types are supported by DataPrep to clean:",
+            [k for k, v in stats.items() if v],
+        )
+    if option == "header":
+        print("Column Headers Cleaning Report:")
+        if stats["cleaned"] > 0:
+            nclnd = stats["cleaned"]
+            pclnd = round(nclnd / old_stat * 100, 2)
+            print(f"\t{nclnd} values cleaned ({pclnd}%)")
+        else:
+            print("No Headers Cleaned.")
+    if option == "duplicate":
+        print("Number of Entries Cleaning Report:")
+        if stats["cleaned"] > 0:
+            nclnd = stats["cleaned"]
+            pclnd = round(nclnd / old_stat * 100, 2)
+            print(f"\t{nclnd} entries dropped ({pclnd}%)")
+        else:
+            print("No Duplicated Entries Cleaned.")
+    if option == "memory":
+        print("Downcast Memory Report:")
+        if stats["cleaned"] > 0:
+            nclnd = stats["cleaned"]
+            pclnd = round(nclnd / old_stat * 100, 2)
+            print(f"\tMemory reducted from {old_stat} to {nclnd}. New size: ({pclnd}%)")
+        else:
+            print("Downcast Memory not Performed.")
+
```

Comment:
Maybe it's better to add an illegal checking here, to avoid illegal call

Reply:
Thanks Yi. Sorry I did not get it, what type of illegal call it can possibly be?

Reply:
What I mean is that input `option` parameter can be an illegal string.

Reply:
I am not very sure whether it could happen, as the `_create_report()` function is only callable inside `clean_df()` by the `if report: (line 131)` condition. But users might directly call this function accidentally and receive unexpected errors. So I am OK to add a check condition inside this function.

By the way, it just reminds me that users could get confused by this `_create_report()` with the other report functions in `utils.py`? Is it necessary to rename this function to `_create_report_clean_df()`?

Reply:
Yes, what I'm referring to is potential accidental touch by users. But I'm OK with either way so just let me know your decision. For the function name, do you think `create_report_df()` or `create_df_report()` will work?

Reply:
I feel like ` create_report_df()` works. We can also listen to Danrui's opinion!

Reply:
Fixed in the newest commit!